### PR TITLE
add t5 stack that can function as either the encoder or decoder of a t5 model

### PR DESCRIPTION
**Pull request by pmabbo13**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #1829
* __->__ #1828
* #1827
* #1826
* #1833
* #1832
* #1831
* #1830
* #1825

### CODE REVIEW

File: torchtext/prototype/t5/modules.py [hunk 1]
```diff
@@ -724,3 +724,108 @@ def _ca_block(
     def _ff_block(self, x: Tensor) -> Tensor:
         x = self.linear2(self.dropout2(self.activation(self.linear1(x))))
         return self.dropout3(x)
+
+
+class T5Stack(nn.Module):
```
Comment (Nayef211) [inline torchtext/prototype/t5/modules.py:729]: include permalink to component from HF

**Review by Nayef211 — APPROVED** [review]
Leave comment with differences if any from original implementation

**Comment by pmabbo13** [comment]
> Leave comment with differences if any from original implementation

updated!

File: torchtext/prototype/t5/modules.py [hunk 2]
```diff
@@ -612,3 +612,113 @@ def _ca_block(
...
+
+    def forward(
+        self,
+        tgt: Tensor,
+        memory: Tensor = None,
+        tgt_mask: Optional[Tensor] = None,
+        memory_mask: Optional[Tensor] = None,
+        tgt_key_padding_mask: Optional[Tensor] = None,
+        memory_key_padding_mask: Optional[Tensor] = None,
+    ) -> Tensor:
```
Comment (parmeet) [inline torchtext/prototype/t5/modules.py:687]: Looks like annotation doesn't match the output?

**Review by parmeet — APPROVED** [review]
LGTM!

**Pull request merged**